tools/package-tools/build-release.ts

```diff
@@ -26,14 +29,73 @@ export function composeRelease(packageName: string) {
-  copyFiles(bundlesDir, `${packageName}.umd?(.min).js?(.map)`, join(releasePath, 'bundles'));
+  copyFiles(bundlesDir, `*.umd?(.min).js?(.map)`, join(releasePath, 'bundles'));
   copyFiles(bundlesDir, `${packageName}?(.es5).js?(.map)`, join(releasePath, '@angular'));
+  copyFiles(join(bundlesDir, packageName), '**', join(releasePath, '@angular', packageName));
   copyFiles(projectDir, 'LICENSE', releasePath);
   copyFiles(packagesDir, 'README.md', releasePath);
   copyFiles(sourcePath, 'package.json', releasePath);
 
   updatePackageVersion(releasePath);
-  createTypingsReexportFile(releasePath, packageName);
-  createMetadataReexportFile(releasePath, packageName);
+  createTypingsReexportFile(releasePath, './typings/index', packageName);
+  createMetadataReexportFile(releasePath, './typings/index', packageName);
+
+  if (options.useSecondaryEntryPoints) {
+    createFilesForSecondaryEntryPoint(packageName, packagePath, releasePath);
+  }
+
   addPureAnnotationsToFile(join(releasePath, '@angular', `${packageName}.es5.js`));
 }
+
+/**
+ * Creates files necessary for a secondary entry-point.
+ * @param packageName The name of the package for which to create entry-point files.
+ * @param packagePath
+ * @param releasePath The path to the release package.
+ */
+function createFilesForSecondaryEntryPoint(packageName: string, packagePath: string, releasePath: string) {
+  getSecondaryEntryPointsForPackage(packageName).forEach(entryPointName => {
+    // Create a directory in the root of the package for this entry point that contains
+    // * A package.json that lists the different bundle locations
+    // * An index.d.ts file that re-exports the index.d.ts from the typings/ directory
+    // * A metadata.json re-export for this entry-point's metadata.
+    const entryPointDir = join(releasePath, entryPointName);
+    mkdirpSync(entryPointDir);
+    createEntryPointPackageJson(entryPointDir, packageName, entryPointName);
+
+    // Copy typings and metadata from tsc output location into the entry-point.
+    copyFiles(
+        join(packagePath, entryPointName),
+        '**/*.+(d.ts|metadata.json)',
+        join(entryPointDir, 'typings'));
+
+    // Create a typings and a metadata re-export within the entry-point to point to the
+    // typings we just copied.
+    createTypingsReexportFile(entryPointDir, `./typings/index`, 'index');
+    createMetadataReexportFile(entryPointDir, `./typings/index`, 'index');
+
+    // Finally, create both a d.ts and metadata file for this entry-point in the root of
+    // the package that re-exports from the entry-point's directory.
+    createTypingsReexportFile(releasePath, `./${entryPointName}/index`, entryPointName);
+    createMetadataReexportFile(releasePath, `./${entryPointName}/index`, entryPointName);
+  });
+}
+
+
+/** Creates a package.json file for a secondary entry-point within a package. */
+function createEntryPointPackageJson(destDir: string, packageName: string, entryPointName: string) {
```

Comment:
This should also go into a separate file as every other "helper" function we recently moved out (fine as follow-up)